src/hipscat_import/run_import.py

```diff
@@ -1,54 +1,76 @@
 """Import a set of non-hipscat files using dask for parallelization"""
 
+
 import hipscat.io.write_metadata as io
 import numpy as np
 from dask.distributed import Client, as_completed
 from hipscat import pixel_math
 from tqdm import tqdm
 
 import hipscat_import.map_reduce as mr
+import hipscat_import.resume_files as rf
 from hipscat_import.arguments import ImportArguments
 
 
 def _map_pixels(args, client):
     """Generate a raw histogram of object counts in each healpix pixel"""
 
+    raw_histogram = rf.read_histogram(args.tmp_path, args.highest_healpix_order)
+    if rf.is_mapping_done(args.tmp_path):
+        return raw_histogram
+
+    mapped_paths = rf.read_mapping_keys(args.tmp_path)
+
     futures = []
     for i, file_path in enumerate(args.input_paths):
+        if file_path in mapped_paths:
+            continue
```

Comment:
After practicing lots of interview questions I feel I have to point out mapped_paths could be made into a set to make this check quicker but it's probably a pointless optimization

Reply:
It's also a 4 character change, and the input list should already be sorted. So seems like it won't be expensive to turn into a set.